File: src/main/java/org/shark/example/domain/user/inbound/web/UserController.java
package org.shark.example.domain.user.inbound.web;

import lombok.RequiredArgsConstructor;
import org.shark.example.domain.user.application.RegisterUserService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@RequestMapping("/user")
public class UserController {

    private final RegisterUserService registerUserService;

    @PostMapping
    public void register(@RequestParam String name, @RequestParam String email) {
        registerUserService.register(name, email);
    }
}

File: src/main/kotlin/org/shark/example/domain/user/inbound/web/UserController.kt
package org.shark.example.domain.user.inbound.web

import org.shark.example.domain.user.application.RegisterUserService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/user")
class UserController(
    private val registerUserService: RegisterUserService
) {
    @PostMapping
    fun register(@RequestParam name: String, @RequestParam email: String) {
        registerUserService.register(name, email)
    }
}
